llvm/lib/Target/AArch64/AArch64FrameLowering.cpp

```diff
@@ -2736,13 +2849,21 @@ StackOffset AArch64FrameLowering::resolveFrameOffsetReference(
       "In the presence of dynamic stack pointer realignment, "
       "non-argument/CSR objects cannot be accessed through the frame pointer");
 
+  bool FPAfterSVECalleeSaves =
+      isTargetWindows(MF) && AFI->getSVECalleeSavedStackSize();
+
   if (isSVE) {
+    assert(-ObjectOffset > (int64_t)AFI->getSVECalleeSavedStackSize() &&
+           "Math isn't correct for CSRs with FPAfterSVECalleeSaves");
```

Comment:
This assertion should be moved under the `FPAfterSVECalleeSaves` check below to prevent the crash on non-Windows targets. The maths may also need fixing for Windows (that might also hit this), but that's a separate thing.